examples/lb3-application/README.md

```diff
@@ -337,13 +337,203 @@ $ npm start
 Load [http://localhost:3000/](http://localhost:3000/) on your browser. This will
 load the Express app, with mounted LB3 and LB4 applications.
 
-### Need help?
+## Running LB3 tests from LB4
+
+You can run tests in an LoopBack 3 application from the LoopBack 4 application
+it mounted on with command `npm tests`.
+
+We want the LoopBack 3 tests to use the LoopBack 4 server rather than the
+LoopBack 3 application. The following guide shows how to run
+
+- acceptance-level tests making HTTP calls to invoke application logic. e.g.
+  `POST /users/login`
+- integration-level tests that are using JS API to call application logic. e.g.
+  `MyModel.create()`
+
+### Adding LB3 Test Path in Command
+
+In order to run LoopBack 3's tests from their current folder, add LB3 tests'
+path to `test` entry in package.json:
+
+- `"test": "lb-mocha \"dist/**tests**/\*_/_.js\" \"lb3app/test/\*.js\""`
+
+In this case, the test folder is
+[`/lb3app/test`](https://github.com/strongloop/loopback-next/tree/spike/lb3test/examples/lb3-application/lb3app/test)
+from the root of the LoopBack 4 project.
+
+This will run LoopBack 4 tests first then LoopBack 3 tests.
+
+_To emphasize the setup steps and separate them from the test case details, all
+the comprehensive test code are extracted into function `runTests`._
+
+### Running Acceptance Tests
+
+First, move any LoopBack 3 test dependencies to `package.json`'s devDependencies
+and run:
+
+```sh
+npm install
+```
+
+In your test file:
+
+1. When launch the LoopBack 4 Express server
+
+- 1.1 Update to use the LB4 Express server when doing requests:
+
+  ```ts
+  // can use lb4's testlab's supertest as the dependency is already installed
+  const {supertest} = require('@loopback/testlab');
+  const assert = require('assert');
+  const should = require('should');
+  const {ExpressServer} = require('../../dist/server');
+
+  let app;
+
+  function jsonForExpressApp(verb, url) {
+    // use the express server, it mounts LoopBack 3 apis to
+    // base path '/api'
+    return supertest(app.server)
+      [verb]('/api' + url)
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/);
+  }
+  ```
+
+- 1.2 Boot and start the LB4 app in your before hook, and stop the app in the
+  after hook:
+
+  ```ts
+  describe('LoopBack 3 style tests - Launch Express server', function () {
+    before(async function () {
+      app = new ExpressServer();
+      await app.boot();
+      await app.start();
+    });
+
+    after(async () => {
+      await app.stop();
+    });
+
+    // your tests here
+    runTests();
+  });
+  ```
+
+2. When launch the LoopBack 4 application
+
+- 2.1 Update to use the LB4 Express server when doing requests:
```

Comment:
A question: what's the difference of running lb3 tests with LB4 Express server or with LB4 Express server?

Reply:
good question, this example demos two scenarios to mount a LB3 app:

- `application.ts` has a LB 4 application `CoffeeShopApplication` mounting the LB3 app, see [code](https://github.com/strongloop/loopback-next/blob/master/examples/lb3-application/src/application.ts#L28)
- `server.ts` has an Express server mounts the coffeeshop app including the LB3 app, see [code](https://github.com/strongloop/loopback-next/blob/master/examples/lb3-application/src/server.ts#L37)

They setup the server and retrieve bindings differently, the PR has guide for both.

Reply:
And I also corrected the doc's titles.